app/src/common/components/Button.tsx

```diff
@@ -48,12 +47,25 @@ export const EditedButton = withStyles((theme) => ({
   },
 }))(MuiButton);
 
-const Button = ({ children, ...buttonProps }: ButtonProps): JSX.Element => {
-  return (
-    <EditedButton disableElevation {...buttonProps}>
-      <Typography>{children}</Typography>
-    </EditedButton>
-  );
-};
+type ButtonProps<C extends React.ElementType> = MuiButtonProps<
+  C,
+  { component?: C }
+>;
+
+const Button = React.forwardRef(
+  <C extends React.ElementType>(
+    { children, ...buttonProps }: ButtonProps<C>,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ref: any
```

Comment:
`React.RefObject<HTMLElement>` ?

Reply:
I tried many things... Problem is that ref type depends on `C` type value and I did not manage de find a solution.

I tried this typing, but got this TS error `Type 'RefObject<HTMLElement> | null | undefined' is not assignable to type '((instance: HTMLButtonElement | null) => void) | RefObject<HTMLButtonElement> | null | undefined'. `